### PR TITLE
Drops Scala 2.11 support on master

### PR DESCRIPTION
**Pull request by kailuowang**
~Another PR to add the scala 2.11 branch coming soon~

PR to add the scala_2.11 branch  #3052

### CODE REVIEW

File: tests/src/test/scala/cats/tests/EitherSuite.scala [hunk 1]
```diff
@@ -346,7 +346,7 @@ class EitherSuite extends CatsSuite {
   test("raiseOrPure syntax consistent with fromEither") {
     val ev = ApplicativeError[Validated[String, *], String]
     forAll { (fa: Either[String, Int]) =>
-      fa.raiseOrPure[Validated[String, *]] should ===(ev.fromEither(fa))
+      fa.liftTo[Validated[String, *]] should ===(ev.fromEither(fa))
```
Comment (travisbrown) [inline tests/src/test/scala/cats/tests/EitherSuite.scala]: I just got bitten by leaving a deprecated method untested, so this makes me a little nervous.

File: tests/src/test/scala/cats/tests/EitherSuite.scala [hunk 2]
```diff
@@ -346,7 +346,7 @@ class EitherSuite extends CatsSuite {
   test("raiseOrPure syntax consistent with fromEither") {
```
Comment (travisbrown) [inline tests/src/test/scala/cats/tests/EitherSuite.scala]: If we're changing method do you mind updating the test description to match?

**Comment by kailuowang** [comment]
@LukaJCB , @travisbrown  got time to take a look at the commit I added to address feedback?

**Comment by travisbrown** [comment]
:+1:

**Pull request merged**